server/src/main/java/org/opensearch/action/admin/indices/cache/clear/ClearIndicesCacheRequest.java

```diff
@@ -120,8 +147,23 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeBoolean(fieldDataCache);
         out.writeStringArrayNullable(fields);
         out.writeBoolean(requestCache);
+        if (out.getVersion().onOrAfter(Version.V_2_12_0)) {
+            out.writeBoolean(requestCacheOnDisk);
+        }
+        if (out.getVersion().onOrAfter(Version.V_2_12_0)) {
+            out.writeBoolean(requestCacheOnHeap);
+        }
         if (out.getVersion().onOrAfter(Version.V_2_8_0)) {
             out.writeBoolean(fileCache);
         }
     }
+
+    public void validateInput() {
+        if (requestCache && requestCacheOnDisk) {
+            throw new IllegalArgumentException("Invalid parameters: cannot have both request and requestCacheOnDisk set to true");
+        }
+        if (requestCache && requestCacheOnHeap) {
+            throw new IllegalArgumentException("Invalid parameters: cannot have both request and requestCacheOnHeap set to true");
+        }
+    }
```

Comment:
Lets remove this for now and keep it consistent with rest of the actions. Move this logic to the actual function.
Doesn't look much useful to me.

Reply:
the validity of the ClearIndicesCacheRequest should be scoped to the ClearIndicesCacheRequest class right ?
the caller of this class i.e RestClearIndicesCacheAction should not be concerned about that.